app/src/main/java/com/example/android/codelabs/paging/ui/SearchRepositoriesActivity.kt

```diff
@@ -103,11 +103,11 @@ class SearchRepositoriesActivity : AppCompatActivity() {
         // Scroll to top when the list is refreshed from network.
         lifecycleScope.launch {
             adapter.loadStateFlow
-                    // Only emit when REFRESH LoadState for RemoteMediator changes.
-                    .distinctUntilChangedBy { it.refresh }
-                    // Only react to cases where Remote REFRESH completes i.e., NotLoading.
-                    .filter { it.refresh is LoadState.NotLoading }
-                    .collect { binding.list.scrollToPosition(0) }
+                // Only emit when REFRESH LoadState for RemoteMediator changes.
```

Comment:
For this you can also try `.drop(1)` to ignore initial `NotLoading` state to handle the `.cachedIn()` / fragment recreate / config change, etc case.